# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegLibWrapper.h
# Commit: c159181ff5d0
# Full Hash: c159181ff5d00814ffa6c1648dc22002b70da9ed
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2020-04-21 22:25:03
# Regressor Bug: 1630754
# File Overlap Count: 1
# Description:
#   Bug 1630754 [Wayland][VA-API] Explicitly configure VADisplay display for VA-API video playback, r=jya
#   
#   Some gfx drivers (especially on a child process) can open X11 VADisplay instead of a Wayland one which leads
#   to HW playback failure. As a solution let's create VADisplay explicitly on top of our wayland display connection.
#   
# ==============================================================================

diff -r 27d3c6f3204e -r c159181ff5d0 dom/media/platforms/ffmpeg/FFmpegLibWrapper.h
--- a/dom/media/platforms/ffmpeg/FFmpegLibWrapper.h	Tue Apr 21 10:50:17 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegLibWrapper.h	Tue Apr 21 10:52:47 2020 +0000
@@ -116,9 +116,9 @@
 #ifdef MOZ_WAYLAND
   const AVCodecHWConfig* (*avcodec_get_hw_config)(const AVCodec* codec,
                                                   int index);
-  int (*av_hwdevice_ctx_create)(AVBufferRef** device_ctx, int type,
-                                const char* device, AVDictionary* opts,
-                                int flags);
+  AVBufferRef* (*av_hwdevice_ctx_alloc)(int);
+  int (*av_hwdevice_ctx_init)(AVBufferRef* ref);
+
   AVBufferRef* (*av_buffer_ref)(AVBufferRef* buf);
   void (*av_buffer_unref)(AVBufferRef** buf);
   int (*av_hwframe_transfer_get_formats)(AVBufferRef* hwframe_ctx, int dir,
@@ -132,12 +132,16 @@
 
   int (*vaExportSurfaceHandle)(void*, unsigned int, uint32_t, uint32_t, void*);
   int (*vaSyncSurface)(void*, unsigned int);
+  int (*vaInitialize)(void* dpy, int* major_version, int* minor_version);
+  int (*vaTerminate)(void* dpy);
+  void* (*vaGetDisplayWl)(struct wl_display* display);
 #endif
 
   PRLibrary* mAVCodecLib;
   PRLibrary* mAVUtilLib;
 #ifdef MOZ_WAYLAND
   PRLibrary* mVALib;
+  PRLibrary* mVALibWayland;
 #endif
 
  private: